hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java

```diff
@@ -74,6 +74,10 @@ public MultiResult() {
         this.results = new ArrayList<T>();
     }
 
+    public MultiResult(List<T> results) {
+        this.results = new ArrayList<T>(results);
```

Comment:
It's was defensively copied with the assumption that the ValueCollector that calls this constructor will be recycled & reused so we didn't want to get this list reset when the Collector was recycled.
But for now, the collector is not reused so I can just take the reference as-is.